src/views/Graph.vue

```diff
@@ -508,16 +1007,108 @@ export default {
     /**
      * Get the nodes binned by cycle point
      *
-     * @param {Object[]} nodes
+     * @param {Node[]} nodes - The graph nodes
      * @returns {{ [dateTime: string]: Object[] }=} mapping of cycle points to nodes
      */
     getCycles (nodes) {
-      if (!this.groupCycle) return
       return nodes.reduce((x, y) => {
         (x[y.tokens.cycle] ||= []).push(y)
         return x
       }, {})
     },
+    /**
+     * Recursive function that adds a subgraph to the dot code
+     *
+     * Terminology:
+     * The pointer is a node object passed to the method
+     * The pointer child is the family that will be grouped
+     * The pointer grandchildren are the nodes that will be included in the grouping
+     *
+     * @param {String[]} dotcode - The array of strings that make up the dot code
+     * @param {Node} pointer - Node object pointer used for recursion to navigate graph tree
+     */
+    addSubgraph (dotcode, pointer) {
+      // If there are no families grouped we dont need to run this code
+      if (!this.groupFamily.length) {
+        return
+      }
+      // We can only group by nodes of type family
+      if (pointer.type !== 'family') {
+        return
+      }
+
+      // The pointer has children
+      // We want to see if we can group each child
+      for (const child of pointer.children) {
+        // The pointer child has children (grandChildren)
+        // These grandChildren are the nodes that will be included in the grouping
+        const grandChildren = this.allChildrenLookUp[child.id]
+        if (!grandChildren) { return }
+        // Some of the nodes may have been collapsed
+        // Work out if any have and store for reference later
+        const removedNodes = new Set()
+        for (const grandChild of grandChildren) {
+          if (this.collapseFamily.includes(grandChild.name)) {
+            for (const child of grandChild.children) {
+              removedNodes.add(child.name)
+            }
+          }
+        }
+        let openedBrackets = false
+        if (
+          // If this pointer has grandchildren
+          grandChildren.length &&
+          // Is the child of the pointer a family that is grouped?
+          // If yes - we want to include in the grouping
+          this.groupFamily.includes(child.node.name) &&
+          // But if its collapsed then we dont want to group it
+          // We dont put boxes around collapsed nodes - design choice
```

Comment:
I'd question that design choice, but I'm guessing that it's been thrashed out?

I think that collapsed nodes should retain the box to indicated that it's not just a single task.